docs/topics/OAuth2.md

```diff
@@ -168,6 +168,37 @@ def refresh_token(refresh_token):
 
 Boom; fresh [access token response](#DOCS_TOPICS_OAUTH2/authorization-code-grant-access-token-response)!
 
+###### Token Revocation Example
+
+To revoke a token, make a `POST` request to the [token revocation URL](#DOCS_TOPICS_OAUTH2/shared-resources-oauth2-urls) with the following parameters:
+
+- `Authorization: Basic <base64 encoded client_id:client_secret>` - your application's client id and client secret, separated by a colon, and base64 encoded as header.
+- `token` - the access token or refresh token to revoke
```

Comment:
It's also possible to pass the client id and secrets in the request body.

For consistency with other examples on this page (like [Autorization Code Grant](https://discord.com/developers/docs/topics/oauth2#authorization-code-grant-redirect-url-example)), maybe we should instead document passing them in the body, and add a note that they can also be passed as basic authentication.

Reply:
I'm not sure why you means by "It's not". Passing the client id/secret in the body definitely work:
![image](https://github.com/discord/discord-api-docs/assets/5215545/67ec10c9-8c35-4287-bee7-261418023026)

Reply:
weird, well, rfc tho 🤷‍♀️

Reply:
Is this documentation for the Discord API or for the original OAuth2 RFC?

Reply:
Checking again the [RFC 6749 Section 2.3.1](https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1), both methods can be supported by the server, but the RFC discourages including the client_id/secret in the body:

> Including the client credentials in the request-body using the two parameters is NOT RECOMMENDED and SHOULD be limited to clients unable to directly utilize the HTTP Basic authentication scheme (or other password-based HTTP authentication schemes).

So for consistency, it may be better to update other examples to use the `Authorization` header instead.

Reply:
I'll wait for staff response on that.
The revocation request layout was given by me by one, hence why I noted it down like that.

Reply:
And splatter, both

Reply:
Basic authentication is already in the docs. 

> You can also pass your `client_id` and `client_secret` as basic authentication with `client_id` as the username and `client_secret` as the password.
> — [access token exchange example](https://discord.com/developers/docs/topics/oauth2#authorization-code-grant-access-token-exchange-example)

I don’t see a need to refer to it again especially as previous examples send the credentials in the request bodies.

Reply:
> You can also pass your client_id and client_secret as basic authentication with client_id as the username and client_secret as the password

It doesn't mention that it has to be base64encoded, so an example on that wouldn't harm

Reply:
And about
> So for consistency, it may be better to update other examples to use the Authorization header instead.

@JustinBeckwith what are your thoughts?

Reply:
I'm not bothered by the inconsistency.  We can always come back around and fix the other example :) For this one, following the spec guidance to pass the CLIENT_ID and CLIENT_SECRET in the authorization header feels right, assuming someone has confirmed that works (all of my code just sends them in the request body because I'm lazy).

Reply:
Tested all routes with the auth header, it works